src/shape/vertex.js

```diff
@@ -803,33 +802,21 @@ function vertex(p5, fn){
    * @param  {Number} x4
    * @param  {Number} y4
    * @param  {Number} z4 z-coordinate of the anchor point.
-   * @chainable
    */
   fn.bezierVertex = function(...args) {
-    p5._validateParameters('bezierVertex', args);
-    if (this._renderer.isP3D) {
-      this._renderer.bezierVertex(...args);
-    } else {
-      if (vertices.length === 0) {
-        p5._friendlyError(
-          'vertex() must be used once before calling bezierVertex()',
-          'bezierVertex'
-        );
-      } else {
-        isBezier = true;
-        const vert = [];
-        for (let i = 0; i < args.length; i++) {
-          vert[i] = args[i];
-        }
-        vert.isVert = false;
-        if (isContour) {
-          contourVertices.push(vert);
-        } else {
-          vertices.push(vert);
-        }
+    if (args.length === 2 * 3 || args.length === 3 * 3) {
+      // Handle the legacy case where all bezier control points are provided
+      // at once. We'll translate them into 3 individual calls.
+      const stride = args.length / 3;
+
+      const prevOrder = this._renderer.bezierOrder();
```

Comment:
Can you explain what's happening here? Seems like you're saving the order so you can reset it back to its original value, but it doesn't seem necessary since the order doesn't get changed. I think I'm misunderstanding something.

Reply:
good catch, it needs to set it to 3 in here!